orttraining/orttraining/core/optimizer/graph_transformer_utils.cc

```diff
@@ -75,6 +76,9 @@ std::vector<std::unique_ptr<GraphTransformer>> GeneratePreTrainingTransformers(
       rule_transformer->Register(make_unique<NonZeroShapeSetter>());
       rule_transformer->Register(make_unique<InsertSoftmaxCrossEntropyLossOutput>());
 
+      // Remove duplicate nodes. Must be applied before any recompute transformations.
+      transformers.emplace_back(onnxruntime::make_unique<CommonSubexpressionEliminationApplyOnce>(compatible_eps));
```

Comment:
we also need to disable the CommonSubexpressionElimination in inference session. 

Maybe with #ifdef ENABLE_TRAINING ？

Reply:
As per offline discussion, submitting this as is, and will figure out recompute interactions later.